woocommerce/woocommerce-ios
# [Mobile Payments] Show and hide the red badge on top of the payments button within the hub menu

Closes: #7360 


### Description

With this PR we implement showing and hiding the badge on the Payments button depending on whether the Payments screen was open or not.
The main challenge was to pass dynamically the badge information into the `HubMenu` view. Before we had an enum with all the button cases, and iterate in the view through them. Now that we need to set the information dynamically because of the badge visibility changes, we migrated the `enum` into a set of `structs` that conform to a `protocol` HubMenuItem.

### Testing instructions


- Make a clean installation.
- Open the app.
- Go to the Menu Hub screen
- The Payments should have a badge on it
- Tap on Payments button
- Go back. The Payments button should not have the badge
- Re open the app. The Payments button should not have the badge

See video below.

### Screenshots


https://user-images.githubusercontent.com/1864060/182610621-4f4fc643-3d1c-4d9c-908f-f3a67538015c.mp4

---
